docker-compose.yml

```diff
@@ -8,21 +8,30 @@ services:
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=postgres
     volumes:
-      - ./postgres:/var/lib/postgresql/data
+      - oss-dashboard-postgres:/var/lib/postgresql/data
+    networks:
+      - db
   dashboard:
     build:
       context: ./
       args:
         - http_proxy
         - https_proxy
     environment:
-      - GH_ACCESS_TOKEN=
-      - OCTOKIT_API_ENDPOINT= # if you use oss-dashboard to GitHub Enterprise, specify GHE api endpoint (ex. https://github.mycompany.com/api/v3)
+      - GH_ACCESS_TOKEN
+      - OCTOKIT_API_ENDPOINT # if you use oss-dashboard to GitHub Enterprise, specify GHE api endpoint (ex. https://github.mycompany.com/api/v3)
+    entrypoint: /wait-for db:5432 -t 10 -- bundle exec ruby
     command: refresh-dashboard.rb config-dashboard.yaml
     volumes:
       - ./config-dashboard.yaml:/oss-dashboard/config-dashboard.yaml
       # - ./config-github.yaml:/oss-dashboard/config-github.yaml
       - ./data:/oss-dashboard/data
       - ./html:/oss-dashboard/html
-    links:
+    networks:
```

Comment:
I'm curious, why the network declaration instead of the link? Are the below sections (both `networks` and `volumes`) valid as mostly-empty blocks?

Reply:
The `links` are a mechanism of Docker-Compose v2 and allow to connect containers on your local machine only. `networks` allow you to spread your containers across different machines using an overlay network. When running the Compose setup on your local machine you won't see any difference, but I changed it to make the compose file more future proof and make it easier to use it in a Docker Swarm.
The mostly empty blocks are valid, though looking a bit strange. They make use of some defaults in Docker itself. In this case I wanted to keep the compose file as declarative as possible.

Reply:
I see, thanks for explaining. 👍

Reply:
You're welcome :-)